src/java.desktop/share/classes/java/awt/geom/Path2D.java

```diff
@@ -2129,6 +2081,140 @@ public final Rectangle getBounds() {
         return getBounds2D().getBounds();
     }
 
+    /**
+     * {@inheritDoc}
+     * @since 1.6
+     */
+    public Rectangle2D getBounds2D() {
+        return getBounds2D(getPathIterator(null));
+    }
+
+    /**
+     * Returns a high precision bounding box of the specified PathIterator.
+     * <p>
+     * This method provides a basic facility for implementors of the {@link Shape} interface to
+     * implement support for the {@link Shape#getBounds2D()} method.
+     * </p>
+     * @return an instance of {@code Rectangle2D} that is a high-precision bounding box of the
+     *         {@code PathIterator}.
+     * @see Shape#getBounds2D()
+     */
+    public static Rectangle2D getBounds2D(PathIterator pi) {
+        // define x and y parametric coefficients where:
+        // x(t) = x_coeff[0] + x_coeff[1] * t + x_coeff[2] * t^2 + x_coeff[3] * t^3
+        double[] x_coeff = new double[4];
+        double[] y_coeff = new double[4];
+
+        double[] coords = new double[6];
+        double[] tExtrema = new double[3];
+        boolean isEmpty = true;
+        double leftX = 0.0;
+        double rightX = 0.0;
+        double topY = 0.0;
+        double bottomY = 0.0;
+        double lastX = 0.0;
+        double lastY = 0.0;
+
+        pathIteratorLoop : while (!pi.isDone()) {
+            int type = pi.currentSegment(coords);
+            pi.next();
+            double endX, endY;
+            switch (type) {
+                case PathIterator.SEG_MOVETO, PathIterator.SEG_LINETO:
+                    endX = coords[0];
+                    endY = coords[1];
+                    break;
+                case PathIterator.SEG_QUADTO:
+                    endX = coords[2];
+                    endY = coords[3];
+                    break;
+                case PathIterator.SEG_CUBICTO:
+                    endX = coords[4];
+                    endY = coords[5];
+                    break;
+                default:
```

Comment:
add explicitely the SEG_CLOSE case (skip = continue) before the default case (impossible case)